solidity/src/hyperkzg/HyperKZGHelpers.pre.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: UNLICENSED
+// This is licensed under the Cryptographic Open Software License 1.0
+pragma solidity ^0.8.28;
+
+// assembly only constants
+/* solhint-disable no-unused-import */
+import {
+    FREE_PTR,
+    MODULUS,
+    MODULUS_MINUS_ONE,
+    MODULUS_MASK,
+    WORD_SIZE,
+    WORDX2_SIZE,
+    WORDX3_SIZE,
+    WORDX6_SIZE,
+    HYPER_KZG_INCONSISTENT_V
+} from "../base/Constants.sol";
+/* solhint-enable no-unused-import */
+
+/// @title HyperKZGHelpers
+/// @dev Library providing helper functions for the HyperKZG polynomial commitment verifier.
+library HyperKZGHelpers {
+    /// @notice Runs the Fiat-Shamir transcript to generate challenges for the HyperKZG proof
+    /// @custom:as-yul-wrapper
+    /// #### Wrapped Yul Function
+    /// ##### Signature
+    /// ```yul
+    /// run_transcript(com_ptr, v_ptr, w_ptr, transcript_ptr, ell) -> r, q, d
+    /// ```
+    /// ##### Parameters
+    /// * `com_ptr` - the calldata pointer to the beginning of the data in `__com`
+    /// * `v_ptr` - the calldata pointer to the beginning of the data in `__v`
+    /// * `w_ptr` - the calldata pointer to the beginning of the data in `__w`
+    /// * `transcript_ptr` - the memory pointer to the transcript word
+    /// @dev Processes prover messages to generate random challenges
```

Comment:
it would be nice to have the gas cost for each of these calls documented above the method